File: my_flutter_order_management_system_app/lib/data/repository/data_repository.dart
import '../api/api_service.dart';
import '../models/order.dart';
import '../models/product.dart';

class DataRepository {
  final ApiService apiService = ApiService();

  Future<List<Order>> getOrders() async {
    try {
      final response = await apiService.getOrders();
      return response.data.orders;
    } catch (e) {
      throw Exception(e.toString());
    }
  }

  Future<void> deleteOrder(String id) async {
    try {
      await apiService.deleteOrder(id);
    } catch (e) {
      throw Exception(e.toString());
    }
  }

  Future<List<Customer>> getCustomers({String? name}) async {
    try {
      final response = await apiService.getCustomers(name: name);
      return response.data.customers;
    } catch (e) {
      throw Exception(e.toString());
    }
  }

  Future<void> deleteCustomer(String id) async {
    try {
      await apiService.deleteCustomer(id);
    } catch (e) {
      throw Exception(e.toString());
    }
  }

  Future<Customer> updateCustomer(String id, Map<String, dynamic> data) async {
    try {
      return await apiService.updateCustomer(id, data);
    } catch (e) {
      throw Exception(e.toString());
    }
  }

  Future<List<Product>> getProducts({String? name}) async {
    try {
      final response = await apiService.getProducts(name: name);
      return response.data.products;
    } catch (e) {
      throw Exception(e.toString());
    }
  }

  Future<void> deleteProduct(String id) async {
    try {
      await apiService.deleteProduct(id);
    } catch (e) {
      throw Exception(e.toString());
    }
  }

  Future<Product> updateProduct(String id, Map<String, dynamic> data) async {
    try {
      return await apiService.updateProduct(id, data);
    } catch (e) {
      throw Exception(e.toString());
    }
  }
}

File: MyApp/app/src/main/java/com/example/myapp/data/repository/DataRepository.kt
package com.example.myapp.data.repository

import com.example.myapp.data.api.RetrofitClient
import com.example.myapp.data.model.Category
import com.example.myapp.data.model.Customer
import com.example.myapp.data.model.Order
import com.example.myapp.data.model.Product

class DataRepository {
    private val apiService = RetrofitClient.apiService

    suspend fun getOrders(): Result<List<Order>> {
        return try {
            val response = apiService.getOrders()
            if (response.code == 200) {
                Result.success(response.data.orders)
            } else {
                Result.failure(Exception(response.description))
            }
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    suspend fun getCategories(): Result<List<Category>> {
        return try {
            val response = apiService.getCategories()
            if (response.code == 200) {
                Result.success(response.data)
            } else {
                Result.failure(Exception(response.description))
            }
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    suspend fun getCustomers(): Result<List<Customer>> {
        return try {
            val response = apiService.getCustomers()
            if (response.code == 200) {
                Result.success(response.data.customers)
            } else {
                Result.failure(Exception(response.description))
            }
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    suspend fun getProducts(): Result<List<Product>> {
        return try {
            val response = apiService.getProducts()
            if (response.code == 200) {
                Result.success(response.data.products)
            } else {
                Result.failure(Exception(response.description))
            }
        } catch (e: Exception) {
            Result.failure(e)
        }
    }
}